contributor-docs/migrating-to-css-modules.md

```diff
@@ -0,0 +1,69 @@
+# Primer React Styled Components to CSS Modules Refactoring Guide
+
+This guide outlines the steps to follow when refactoring Primer React components from Styled Components to CSS Modules.
+
+---
+
+## Table of Contents
+
+- [Primer React Styled Components to CSS Modules Refactoring Guide](#primer-react-styled-components-to-css-modules-refactoring-guide)
+  - [Table of Contents](#table-of-contents)
+  - [Before Refactoring a Component](#before-refactoring-a-component)
+  - [How to Migrate](#how-to-migrate)
+    - [Refactoring Styled-Components to CSS Modules](#refactoring-styled-components-to-css-modules)
+    - [When Refactoring a Component](#when-refactoring-a-component)
+    - [Testing the Migration](#testing-the-migration)
+  - [Releasing a Component](#releasing-a-component)
+
+---
+
+## Before Refactoring a Component
+
+- **Verify VRT (Visual Regression Testing) Coverage:**
+  - Check for missing VRT coverage. We utilize the VRT tests to make sure we're matching styling in production with current expectations.
+  - Make sure there are `dev` stories for any edge cases spotted in production for the component (ie. sx prop, custom className, styled system attributes).
+- **Ensure All Visual Changes Are Completed:**
+  - Make necessary visual changes **before** creating the CSS Modules refactor PR.
+
+---
+
+## How to Migrate
+
+### Refactoring Styled-Components to CSS Modules
+
+- **Replace `${get('...')}` Syntax:**
+  - Migrate these to CSS Modules classes.
+  - No need for fallbacks in CSS Modules.
+
+### When Refactoring a Component
+
+1. **Check for `className` Prop:**
+   - Ensure the component accepts a `className`.
+2. **Feature Flagging:**
+   - Add a feature flag to toggle the `sx` prop for controlled rollout (staff shipping).
+3. **Create CSS Module:**
+   - Add a corresponding `{Component}.module.css` file.
```

Comment:
Would this be a good place to talk about how we use `:where()` for managing specificity or would this be a different doc?

Reply:
Maybe we could just link to the doc I'm writing here when its done

Reply:
I think that would fall into the css authoring doc https://github.com/primer/react/pull/5037/files